src/Features/CSharp/Portable/InitializeParameter/CSharpAddParameterCheckCodeRefactoringProvider.cs

```diff
@@ -92,5 +92,22 @@ protected override StatementSyntax CreateParameterCheckIfStatement(DocumentOptio
                 statement: ifTrueStatement,
                 @else: null);
         }
+
+        protected override async Task<Document?> TryAddNullCheckToParameterDeclarationAsync(Document document, IParameterSymbol parameter, CancellationToken cancellationToken)
+        {
+            var tree = await document.GetRequiredSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+            var options = (CSharpParseOptions)tree.Options;
+            if (options.LanguageVersion < LanguageVersion.Preview)
```

Comment:
Should we also check whether the parameter already has `!!`?

Reply:
> Should we also check whether the parameter already has `!!`?

I agree with that. This should be added, probably in `ParameterValidForNullCheck` around
https://github.com/dotnet/roslyn/blob/0c54df55b42fa98c81d2b88d9be2f5012ca28721/src/Features/Core/Portable/InitializeParameter/AbstractAddParameterCheckCodeRefactoringProvider.cs#L266